db/data_migration/20160908141608_remove_chapters_from_related_mainstream_urls.rb

```diff
@@ -0,0 +1,121 @@
+# This script removes the last part of a slug for related mainstream content urls
+# that are linking to a chapter of a related mainstream content instead of the
+# related mainstream document.
+# The script also fixes some urls that were prefixed with an http instead of an
+# https such that they can also be migrated - some of the http prefixed urls
+# might have valid base_paths already, in which case we just save the https
+# based url. Some of them might also have chapters in which case we remove the
+# chapters and save.
+
+require 'gds-api-adapters'
+
+def validate_url_content_id(url)
+  url = url.chomp
+  base_path = url[18..-1] if url.length > 17
+  content_id = Whitehall.publishing_api_v2_client.lookup_content_id(base_path: base_path) if base_path
+  content_id
+end
+
+def write_to_file(file_name, params)
+  File.open(file_name, "a+") do |f|
+    params.each do |param|
+      f.write "#{param}, "
+    end
+    f.write "\n"
+  end
+end
+
+def detailed_guide?(id, url)
+  write_to_file("related_mainstream_content.detailed_guide_items_found", [id, url]) if url.start_with?("https://www.gov.uk/guidance/")
+  url.start_with?("https://www.gov.uk/guidance/")
+end
+
+def update_mainstream_url(id, clean_url)
+  detailed_guide = DetailedGuide.find(id)
+  incorrect_url = detailed_guide.related_mainstream_content_url
+  detailed_guide.update_attribute(:related_mainstream_content_url, clean_url)
+  write_to_file("related_mainstream_content.related_content_found_and_updated", [id, incorrect_url, clean_url])
+end
+
+def update_additional_related_mainstream_url(id, clean_url)
+  detailed_guide = DetailedGuide.find(id)
+  incorrect_url = detailed_guide.additional_related_mainstream_content_url
+  detailed_guide.update_attribute(:additional_related_mainstream_content_url, clean_url)
+  write_to_file("related_mainstream_content.related_content_found_and_updated", [id, incorrect_url, clean_url])
+end
+
+def clean_the_url(url)
+  clean_url = url[/(http(s?):\/\/www.gov.uk\/)([^\/]*)/]
+  if !clean_url.nil?
+    clean_url.insert(4, 's') if clean_url.start_with?('http://')
+    clean_url = clean_url.chop if clean_url[-1] == "/"
+  end
+  clean_url
+end
+
+files = %w(related_mainstream_content.original_url_already_perfect
```

Comment:
Bearing in mind that this will be run in Jenkins and it's not easy to get files out of Jenkins, it'd be better to report changes to stdout when this migration runs